modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -1346,6 +1206,166 @@ else if ((objectField.getListTypeDefinitionId() != 0) &&
+				  (objectRelationship.getObjectDefinitionId1() ==
+					  objectDefinition.getObjectDefinitionId())))) {
+
+				ObjectDefinition relatedObjectDefinition =
+					_objectDefinitionLocalService.getObjectDefinition(
+						_getRelatedObjectDefinitionId(
+							objectDefinition, objectRelationship));
+
+				List<LinkedHashMap<String, Object>>
+					nestedObjectEntryPropertiesList =
+						(List<LinkedHashMap<String, Object>>)propertyValue;
+
+				for (Map<String, Object> nestedObjectEntryProperties :
+						nestedObjectEntryPropertiesList) {
+
+					_upsertAndRelateNestedObjectEntry(
+						dtoConverterContext, nestedObjectEntryProperties,
+						objectRelationship, relatedObjectDefinition, false,
+						serviceBuilderObjectEntry);
+				}
+			}
+			else if ((propertyValue instanceof Map) &&
+					 StringUtil.equals(
+						 objectRelationship.getType(),
+						 ObjectRelationshipConstants.TYPE_ONE_TO_MANY) &&
+					 (objectRelationship.getObjectDefinitionId2() ==
+						 objectDefinition.getObjectDefinitionId())) {
+
+				Map<String, Object> nestedObjectEntryProperties =
+					(Map<String, Object>)propertyValue;
+
+				ObjectDefinition relatedObjectDefinition =
+					_objectDefinitionLocalService.getObjectDefinition(
+						_getRelatedObjectDefinitionId(
+							objectDefinition, objectRelationship));
+
+				_upsertAndRelateNestedObjectEntry(
+					dtoConverterContext, nestedObjectEntryProperties,
+					objectRelationship, relatedObjectDefinition, true,
+					serviceBuilderObjectEntry);
+			}
+			else {
+				throw new BadRequestException(
+					"Unable to create nested object entries for object entry " +
+						serviceBuilderObjectEntry.getObjectEntryId());
+			}
+		}
+	}
+
+	private com.liferay.object.model.ObjectEntry _upsertNestedObjectEntry(
+			DTOConverterContext dtoConverterContext,
+			Map<String, Object> nestedObjectEntryProperties,
+			ObjectDefinition relatedObjectDefinition)
+		throws Exception {
+
+		long groupId = getGroupId(
+			relatedObjectDefinition, relatedObjectDefinition.getScope());
+
+		ObjectEntry objectEntry = new ObjectEntry();
+
+		objectEntry.setProperties(nestedObjectEntryProperties);
+
+		if (nestedObjectEntryProperties.containsKey("externalReferenceCode")) {
+			com.liferay.object.model.ObjectEntry serviceBuilderObjectEntry =
+				_objectEntryLocalService.fetchObjectEntry(
+					(String)nestedObjectEntryProperties.get(
+						"externalReferenceCode"),
+					relatedObjectDefinition.getObjectDefinitionId());
+
+			if (serviceBuilderObjectEntry == null) {
+				return _objectEntryService.addObjectEntry(
+					groupId, relatedObjectDefinition.getObjectDefinitionId(),
+					_toObjectValues(
+						groupId, dtoConverterContext.getUserId(),
+						relatedObjectDefinition, objectEntry, 0L,
+						dtoConverterContext.getLocale()),
+					_createServiceContext(
+						nestedObjectEntryProperties,
+						dtoConverterContext.getUserId()));
+			}
+
+			return _objectEntryService.addOrUpdateObjectEntry(
```

Comment:
I don't get the code of this if statement.
I mean, if it comes with an ERC, then exists and we should update the object entry. If not, we should create it. OK, perfect. 
But, you are using the following method `_objectEntryService.addOrUpdateObjectEntry`. I assume that this method creates the object entry if it doesn't exist or updates it in another case. Also, if you see the implementation of that method, it is checking for the ERC and, if it exists, calls the exact method you are using to create: `addObjectEntry`

So, my question is, if the object entry comes with ERC, why do not use directly `_objectEntryService.addOrUpdateObjectEntry` and forget about checking if the object entry exists or not? 
Am I missing something?

Reply:
@LuismiBarcos, you are right, and make sense because it is unnecessary to create all the if statements. Just calling `_objectEntryService.addOrUpdateObjectEntry` is enough